packages/falter-berlin-dynamic-banner/files/uci-defaults/add-banner-cron.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -e
+
+if [ ! -L /etc/banner ]; then
+	mv /etc/banner /etc/banner.bak
+	ln -s -T /tmp/banner /etc/banner
+	#if [ $? ]; then exit 1; fi
+fi
+
+if [ ! "$(crontab -l | grep "refresh_banner.sh")" ]; then
+	echo "*/1 * * * *	/usr/bin/refresh_banner.sh" >> /etc/crontabs/root
```

Comment:
Is it really necessary to run this every minute or even as a cronjob?  

It is possible to have the refresh_banner script run only when someone logs in to the terminal (ssh, serial console). For example, a script can be created called ```/etc/profile.d/myexample.sh``` which would be run whenever someone logs into the system.

Also, ```* * * * *``` is equivalent to ```*/1 * * * *```

Reply:
I didn't know that. Thanks a lot for that tip.

Reply:
in general, placing `refresh_banner.sh` under `/etc/profile.d/` works fine. But that to slow. The refreshing is finished after the banner was printed to the user. So I get constantly the information from last login.

Maybe a better solution would be, to call that script in `/etc/profile` directly, before the banner gets printed.
```bash
#!/bin/sh
[ -e /tmp/.failsafe ] && export FAILSAFE=1

. /usr/bin/refresh_banner.sh
[ -f /etc/banner ] && cat /etc/banner
[ -n "$FAILSAFE" ] && cat /etc/banner.failsafe
```

Reply:
resolved in f8c3f58b2414f1c4a17eaea8301f7b00a1d3768c